server/api/src/rsGeneralAdmin.cpp

```diff
@@ -53,56 +53,55 @@ namespace
+            }
 
-                nlohmann::json zone_report = nlohmann::json::parse(
-                        std::string(static_cast<char *>(bbuf->buf), bbuf->len)
-                )["zones"];
+            nlohmann::json zone_report = nlohmann::json::parse(
+                    std::string(static_cast<char *>(bbuf->buf), bbuf->len)
+            )["zones"];
 
-                const std::string local_zone_name = std::string(getLocalZoneName());
+            const std::string local_zone_name = std::string(getLocalZoneName());
 
-                for ( const auto& zone : zone_report ) {
-                    if ( zone["icat_server"]["service_account_environment"]["irods_zone_name"] == local_zone_name ) {
+            for ( const auto& zone : zone_report ) {
+                if ( zone["icat_server"]["service_account_environment"]["irods_zone_name"] == local_zone_name ) {
 
+                    admins_in_zone.insert(
+                        static_cast<std::string>(zone["icat_server"]["service_account_environment"]["irods_user_name"])
+                    );
+
+                    for ( const auto& server : zone["servers"] ) {
                         admins_in_zone.insert(
-                            static_cast<std::string>(zone["icat_server"]["service_account_environment"]["irods_user_name"])
+                            static_cast<std::string>(server["service_account_environment"]["irods_user_name"])
                         );
-
-                        for ( const auto& server : zone["servers"] ) {
-                            admins_in_zone.insert(
-                                static_cast<std::string>(server["service_account_environment"]["irods_user_name"])
-                            );
-                        }
-
                     }
-                }
 
-                if ( admins_in_zone.find(static_cast<std::string>(_user_name)) != admins_in_zone.end() &&
-                     _new_value != "rodsadmin"
-                   )
-                {
-                    THROW(SYS_NOT_ALLOWED, "Cannot downgrade a rodsadmin running another server in your zone.");
                 }
+            }
 
-                if ( std::string(rsComm.clientUser.userName) == _user_name &&
-                     _new_value != "rodsadmin"
-                   )
-                {
-                   THROW(SYS_NOT_ALLOWED, "Rodsadmin cannot downgrade their own privileges.");
-                }
+            if ( admins_in_zone.find(static_cast<std::string>(_user_name)) != admins_in_zone.end() &&
+                    _new_value != "rodsadmin"
+                )
+            {
+                THROW(SYS_NOT_ALLOWED, "Cannot downgrade a rodsadmin running another server in your zone.");
+            }
+
+            if ( std::string(rsComm.clientUser.userName) == _user_name &&
+                    _new_value != "rodsadmin"
+                )
+            {
+                THROW(SYS_NOT_ALLOWED, "Rodsadmin cannot downgrade their own privileges.");
```

Comment:
"A rodsadmin account cannot downgrade its own privileges."

Do we want to log this as well?

Reply:
For both of these, would it be acceptable to follow the pattern of the other ```throw_if_...``` functions here and add to the connection's error message?

Reply:
I think so.   Consistency is good.